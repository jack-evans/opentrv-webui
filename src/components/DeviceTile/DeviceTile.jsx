import React from 'react'
import { OverflowMenu, OverflowMenuItem } from 'carbon-components-react'

const DeviceTile = ({ name, temp, active }) => {
  return (
    <div className='DeviceTile'>
      <div className='DeviceTile__overflow-container'>
        <OverflowMenu className='DeviceTile__overflow-menu'>
          <OverflowMenuItem />
        </OverflowMenu>
      </div>
      <div className='DeviceTile__name-container'>
        <h3>{name}</h3>
      </div>
      <h3>{temp} degrees</h3>
      <h3>{active.toString()}</h3>
    </div>
  )
}

export default DeviceTile
