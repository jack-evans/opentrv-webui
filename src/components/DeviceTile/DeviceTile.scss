
.DeviceTile {
  background-color: $ui-02;
  border: 1px solid $ui-04;
  height: 250px;
  margin: 40px;
  padding: 10px;
  width: 280px;

  .DeviceTile__overflow-container {
    border-bottom: 1px solid $ui-04;
    display: flex;
    justify-content: flex-end;

    .DeviceTile__overflow-menu {
      outline: none;
    }
  }

  .DeviceTile__name-container {
    display: flex;
    justify-content: flex-start;
    padding-top: 5px;
  }

  .DeviceTile__icon-container {
    align-items: center;
    display: flex;
    height: 120px;
    justify-content: center;

    .DeviceTile__icon-circle {
      align-self: center;
      border: 1px solid $ui-05;
      border-radius: 50%;
      display: flex;
      height: 80px;
      justify-content: center;
      width: 80px;

      div {
        align-self: center;
      }
    }
  }

  .DeviceTile__footer {
    border-top: 1px solid $ui-04;
    display: flex;
    justify-content: space-between;
    padding-top: 10px;

    .DeviceTile__footer-active {
      display: flex;

      .DeviceTile__footer-active-circle {
        align-self: center;
        background-color: $support-02;
        border-radius: 50%;
        height: 20px;
        width: 20px;
      }

      h3 {
        padding-left: 5px;
      }
    }

    .DeviceTile__footer-idle {
      display: flex;

      .DeviceTile__footer-idle-circle {
        align-self: center;
        background-color: $support-01;
        border-radius: 50%;
        height: 20px;
        width: 20px;
      }

      h3 {
        padding-left: 5px;
      }
    }

    .DeviceTile__footer-temp {
      justify-content: flex-end;
    }
  }
}