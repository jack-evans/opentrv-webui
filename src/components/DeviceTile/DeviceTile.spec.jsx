import React from 'react'
import DeviceTile from './DeviceTile'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('DeviceTile.jsx', () => {
  it('renders without crashing', () => {
    expect(shallow(<DeviceTile name='test' temp={26.3} active={false} />)).toMatchSnapshot()
  })
})
