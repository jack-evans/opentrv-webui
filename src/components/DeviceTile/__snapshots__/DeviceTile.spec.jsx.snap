// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceTile.jsx renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeviceTile
    active={false}
    name="test"
    temp={26.3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="DeviceTile__overflow-container"
        >
          <OverflowMenu
            ariaLabel="list of options"
            className="DeviceTile__overflow-menu"
            flipped={false}
            floatingMenu={false}
            iconDescription="open and close list of options"
            iconName="overflow-menu"
            menuOffset={
              Object {
                "left": 60.5,
                "top": 0,
              }
            }
            menuOffsetFlip={
              Object {
                "left": -60.5,
                "top": 0,
              }
            }
            onClick={[Function]}
            open={false}
            tabIndex={0}
          >
            <OverflowMenuItem
              hasDivider={false}
              isDelete={false}
              itemText="Provide itemText"
              onClick={[Function]}
            />
          </OverflowMenu>
        </div>,
        <div
          className="DeviceTile__name-container"
        >
          <h3>
            test
          </h3>
        </div>,
        <h3>
          26.3
           degrees
        </h3>,
        <h3>
          false
        </h3>,
      ],
      "className": "DeviceTile",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <OverflowMenu
            ariaLabel="list of options"
            className="DeviceTile__overflow-menu"
            flipped={false}
            floatingMenu={false}
            iconDescription="open and close list of options"
            iconName="overflow-menu"
            menuOffset={
              Object {
                "left": 60.5,
                "top": 0,
              }
            }
            menuOffsetFlip={
              Object {
                "left": -60.5,
                "top": 0,
              }
            }
            onClick={[Function]}
            open={false}
            tabIndex={0}
          >
            <OverflowMenuItem
              hasDivider={false}
              isDelete={false}
              itemText="Provide itemText"
              onClick={[Function]}
            />
          </OverflowMenu>,
          "className": "DeviceTile__overflow-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ariaLabel": "list of options",
            "children": <OverflowMenuItem
              hasDivider={false}
              isDelete={false}
              itemText="Provide itemText"
              onClick={[Function]}
            />,
            "className": "DeviceTile__overflow-menu",
            "flipped": false,
            "floatingMenu": false,
            "iconDescription": "open and close list of options",
            "iconName": "overflow-menu",
            "menuOffset": Object {
              "left": 60.5,
              "top": 0,
            },
            "menuOffsetFlip": Object {
              "left": -60.5,
              "top": 0,
            },
            "onClick": [Function],
            "open": false,
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "hasDivider": false,
              "isDelete": false,
              "itemText": "Provide itemText",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3>
            test
          </h3>,
          "className": "DeviceTile__name-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "h3",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            26.3,
            " degrees",
          ],
        },
        "ref": null,
        "rendered": Array [
          26.3,
          " degrees",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "false",
        },
        "ref": null,
        "rendered": "false",
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="DeviceTile__overflow-container"
          >
            <OverflowMenu
              ariaLabel="list of options"
              className="DeviceTile__overflow-menu"
              flipped={false}
              floatingMenu={false}
              iconDescription="open and close list of options"
              iconName="overflow-menu"
              menuOffset={
                Object {
                  "left": 60.5,
                  "top": 0,
                }
              }
              menuOffsetFlip={
                Object {
                  "left": -60.5,
                  "top": 0,
                }
              }
              onClick={[Function]}
              open={false}
              tabIndex={0}
            >
              <OverflowMenuItem
                hasDivider={false}
                isDelete={false}
                itemText="Provide itemText"
                onClick={[Function]}
              />
            </OverflowMenu>
          </div>,
          <div
            className="DeviceTile__name-container"
          >
            <h3>
              test
            </h3>
          </div>,
          <h3>
            26.3
             degrees
          </h3>,
          <h3>
            false
          </h3>,
        ],
        "className": "DeviceTile",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <OverflowMenu
              ariaLabel="list of options"
              className="DeviceTile__overflow-menu"
              flipped={false}
              floatingMenu={false}
              iconDescription="open and close list of options"
              iconName="overflow-menu"
              menuOffset={
                Object {
                  "left": 60.5,
                  "top": 0,
                }
              }
              menuOffsetFlip={
                Object {
                  "left": -60.5,
                  "top": 0,
                }
              }
              onClick={[Function]}
              open={false}
              tabIndex={0}
            >
              <OverflowMenuItem
                hasDivider={false}
                isDelete={false}
                itemText="Provide itemText"
                onClick={[Function]}
              />
            </OverflowMenu>,
            "className": "DeviceTile__overflow-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "ariaLabel": "list of options",
              "children": <OverflowMenuItem
                hasDivider={false}
                isDelete={false}
                itemText="Provide itemText"
                onClick={[Function]}
              />,
              "className": "DeviceTile__overflow-menu",
              "flipped": false,
              "floatingMenu": false,
              "iconDescription": "open and close list of options",
              "iconName": "overflow-menu",
              "menuOffset": Object {
                "left": 60.5,
                "top": 0,
              },
              "menuOffsetFlip": Object {
                "left": -60.5,
                "top": 0,
              },
              "onClick": [Function],
              "open": false,
              "tabIndex": 0,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "hasDivider": false,
                "isDelete": false,
                "itemText": "Provide itemText",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3>
              test
            </h3>,
            "className": "DeviceTile__name-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "test",
            },
            "ref": null,
            "rendered": "test",
            "type": "h3",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              26.3,
              " degrees",
            ],
          },
          "ref": null,
          "rendered": Array [
            26.3,
            " degrees",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "false",
          },
          "ref": null,
          "rendered": "false",
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
